src/test/java/app/fitbuddy/operation/service/NewUserServiceTest.java

```diff
@@ -0,0 +1,98 @@
+package app.fitbuddy.operation.service;
+
+import app.fitbuddy.dto.ExerciseDto;
+import app.fitbuddy.exception.FitBuddyException;
+import app.fitbuddy.jpa.entity.DefaultExercise;
+import app.fitbuddy.jpa.repository.DefaultExerciseCrudRepository;
+import app.fitbuddy.jpa.service.crud.ExerciseCrudService;
+import app.fitbuddy.testhelper.DefaultExerciseTestHelper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class NewUserServiceTest {
+	private final static int DUMMY_USER_ID = 17;
+	@InjectMocks	NewUserService newUserService;
+	@Mock	DefaultExerciseCrudRepository defaultExerciseCrudRepository;
+	@Mock	ExerciseCrudService exerciseCrudService;
+
+	@Test
+	public void newUser_whenAddsDefaultExercises_shouldFindAllDefaultExercises() {
+		when(defaultExerciseCrudRepository.findAll()).thenReturn(dummyDefaultExercises());
+
+		newUserService.addDefaultExercises(DUMMY_USER_ID);
+
+		verify(defaultExerciseCrudRepository).findAll();
+	}
+
+	@Test
+	public void newUser_whenAddsDefaultExercises_shouldCreateDtoForEachDefaultExercise() {
+		when(defaultExerciseCrudRepository.findAll()).thenReturn(dummyDefaultExercises());
+
+		newUserService.addDefaultExercises(DUMMY_USER_ID);
+
+		verify(exerciseCrudService, times(dummyDefaultExercises().size())).create(any());
+	}
+
+	@Test
+	public void newUser_whenAddsDefaultExercises_shouldCreateCorrespondingExerciseDtoFromDefaultExercises() {
+		when(defaultExerciseCrudRepository.findAll()).thenReturn(dummyDefaultExercises());
+		ArgumentCaptor<ExerciseDto> exerciseDtoCaptor = ArgumentCaptor.forClass(ExerciseDto.class);
+
+		newUserService.addDefaultExercises(DUMMY_USER_ID);
+
+		verify(exerciseCrudService, times(dummyDefaultExercises().size())).create(exerciseDtoCaptor.capture());
+		List<ExerciseDto> capturedExerciseDto = exerciseDtoCaptor.getAllValues();
+		dummyDefaultExercises().forEach(
+				dummyDefaultExercise -> {
+					assertTrue(hasCorrespondingDto(capturedExerciseDto, dummyDefaultExercise));
+				}
+		);
+	}
+
+	@Test
+	public void newUser_whenAppUserIsNull_shouldThrowException() {
+		assertThrows(FitBuddyException.class, () -> {
+			newUserService.addDefaultExercises(null);
+		});
+	}
+
+	@Test
+	public void newUser_whenAppUserIsNegative_shouldThrowException() {
+		assertThrows(FitBuddyException.class, () -> {
+			newUserService.addDefaultExercises(-7);
+		});
+	}
```

Comment:
We could shorten this part by using `@ParameterizedTest`. Then use `@NullSource` annotation to provide a null value and use `@ValueSource` to provide the exact values.

```
@ParameterizedTest
@NullSource
@ValueSource(ints = { -7 })
public void example(Integer input) { ...
```